testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/resource/constructor/ResourceConstructorParamsTest.java

```diff
@@ -0,0 +1,72 @@
+package org.jboss.resteasy.test.resource.constructor;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.logging.Logger;
+import org.jboss.resteasy.client.jaxrs.ResteasyClient;
+import org.jboss.resteasy.test.resource.constructor.resource.ConstructorNoParamsResource;
+import org.jboss.resteasy.test.resource.constructor.resource.ConstructorParamsMixedResource;
+import org.jboss.resteasy.utils.PortProviderUtil;
+import org.jboss.resteasy.utils.TestUtil;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.core.Response;
+
+
+@RunWith(Arquillian.class)
+@RunAsClient
+public class ResourceConstructorParamsTest {
+    protected static final Logger logger = Logger.getLogger(
+            ResourceConstructorParamsTest.class.getName());
+
+    static ResteasyClient client;
+
+    @Deployment
+    public static Archive<?> deploySimpleResource() {
+        WebArchive war = TestUtil.prepareArchive(
+                ResourceConstructorParamsTest.class.getSimpleName());
+        war.addClass(ConstructorParamsMixedResource.class);
+        war.addClass(ConstructorNoParamsResource.class);
+        return TestUtil.finishContainerPrepare(war, null);
+    }
+
+    @Before
+    public void init() {
+        client = (ResteasyClient)ClientBuilder.newClient();
+    }
+
+    @After
+    public void after() throws Exception {
+        client.close();
+    }
+
+    private String generateURL(String path) {
+        return PortProviderUtil.generateURL(path,
+                ResourceConstructorParamsTest.class.getSimpleName());
+    }
+
+    @Test
+    public void mixedTest() {
+        Response response = client.target(generateURL("/mixed/get"))
+                .request()
+                .get();
+        Assert.assertEquals("Incorrect status code", 500,
```

Comment:
@rsearls Thank you, my previous comments were fixed, server returns correct response code new. But I'm looking to the test and it seems that this testcase is unrelated with original issue (RESTEASY-1364). This testcases tests "500 response code behaviour", that I mentioned in my [previous comment](https://github.com/resteasy/Resteasy/pull/1873/#pullrequestreview-206199383), but this issue needs "400/404 response code behaviour" testcase. Can you provide correct testcase?

Reply:
Done